### PR TITLE
JDK-8304036: Use CommandLine class from shared module

### PR DESCRIPTION
**Pull request by sormuras**
This pull request addresses the open ends left by [JDK-8236919](https://bugs.openjdk.org/browse/JDK-8236919):
- #11272

Changes:
- [x] Extend list of targeted exports of `jdk.internal.opt/jdk.internal.opt` to `jdk.compiler` and `jdk.javadoc`
- [x] Use shared `CommandLine.java` in `jdk.compiler` module
- [x] Use shared `CommandLine.java` in `jdk.javadoc` module
- [x] Remove `CommandLine.java` from `jdk.compiler` module

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304036](https://bugs.openjdk.org/browse/JDK-8304036): Use CommandLine class from shared module

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [7e05ad8d](https://git.openjdk.org/jdk/pull/12997/files/7e05ad8d2d4abbbd3fcbc27bdfe7a7f18fe60b31)
 * [Mandy Chung](https://openjdk.org/census#mchung) (@mlchung - **Reviewer**) ⚠️ Review applies to [26d14f82](https://git.openjdk.org/jdk/pull/12997/files/26d14f8261d4f9307b0edd3709056452878a77c2)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/12997/head:pull/12997` \
`$ git checkout pull/12997`

Update a local copy of the PR: \
`$ git checkout pull/12997` \
`$ git pull https://git.openjdk.org/jdk.git pull/12997/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12997`

View PR using the GUI difftool: \
`$ git pr show -t 12997`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12997.diff">https://git.openjdk.org/jdk/pull/12997.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/12997#issuecomment-1465700673)

### CODE REVIEW

**Comment by sormuras** [comment]
Converted this PR to draft mode until boot JDK is updated to 20.

https://github.com/openjdk/jdk/blob/master/make/conf/github-actions.conf

**Review by jonathan-gibbons — APPROVED** [review]

File: src/jdk.internal.opt/share/classes/module-info.java [hunk 1]
```diff
@@ -30,5 +30,5 @@
  */
 module jdk.internal.opt {
     exports jdk.internal.joptsimple to jdk.jlink, jdk.jshell;
-    exports jdk.internal.opt to jdk.jartool, jdk.jlink, jdk.jpackage;
+    exports jdk.internal.opt to jdk.compiler, jdk.jartool, jdk.javadoc, jdk.jlink, jdk.jpackage;
```
Comment (mlchung) [inline src/jdk.internal.opt/share/classes/module-info.java:33]: Nit: line break at each module is easier to read and avoid long line like:
   https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/module-info.java#L149

Comment (sormuras) [reply src/jdk.internal.opt/share/classes/module-info.java:33]: Yes, that's much nicer. Will reflow all other export directives as well.

**Comment by sormuras** [comment]
Converted this PR to draft mode again: interim langtools are not happy, yet.

**Review by mlchung — APPROVED** [review]

**Review by mlchung — APPROVED** [review]

**Pull request closed without merging**